pkg/kube/kube_handler.go

```diff
@@ -474,8 +479,15 @@ func (a *userHandler) CreateAppRepository(appRepoBody io.ReadCloser, requestName
 	}
 
 	repoSecret := secretForRequest(appRepoRequest, appRepo)
+	if len(appRepoRequest.AppRepository.AuthRegCreds) > 0 {
+		repoSecret, err = a.GetSecret(appRepoRequest.AppRepository.AuthRegCreds, requestNamespace)
```

Comment:
This line seems to be clobbering the `repoSecret` var which could already have a value? Should the above condition include `repoSecret == nil && ...` or otherwise be an if/else so that we're not clobbering the value?

It looks like what you're wanting to do is something like: if the request includes a reference to a secret, use that, otherwise use the secret from the request itself (if present). If so, perhaps this should be in `secretForRequest`?

Reply:
> If so, perhaps this should be in secretForRequest?

Yes, I think that's better.